### ISSUE TITLE
create a standard set of benchmarks for manual compactions

### ISSUE THREAD

**Issue by bananabrick** [cockroachdb/pebble#1644]
Time taken by a manual compaction to de-invert an lsm is unknown. Some factors which may influence this time are:

1. File sizes
2. Number of keys per file
3. Number of files which have range delete blocks
4. Amount of file overlap
5. Overall size of L0
6. Number of sublevels in L0

It'll be nice to have a standard set of roachperf benchmark for a few variations of inverted lsms to figure out time taken to de-invert L0 for various L0 configurations. The results will allow us to make better recommendations during on-call emergencies.

**Comment by nicktrav** [comment]
One thing we could consider is concocting something like a “bad LSM” state as a test / benchmark fixture (and storing somewhere, or building the generator into the pebble binary itself, like we do for `cockroach workload`). A few different types of “bad LSMs” - classic inversion (large L0 sublevel count / size); the newer single L0 sublevel / large size type - would be interesting.

These fixtures could then be used to build something like a consistent / repeatable nightly benchmark.

The results could also be surfaced on the Pebble nightly benchmarks page for tracking improvements over time.

**Comment by bananabrick** [comment]
Good idea! Although, this kind of thing might be better as a weekly benchmark.
